aurora/core.py

```diff
@@ -1189,73 +1339,341 @@ def check_conservation(self, plot=True, axs=None, plot_resolutions=False):
+            ax1[0, 0].plot(time_average, reservoirs["net_plasma_flow"], label="Net sum")
+            ax1[0, 0].set_ylabel('[$s^{-1}$]')
+
+            if self.namelist["phys_volumes"]:
+                ax1[0, 1].plot(time_average, reservoirs["particle_density_in_plasma"],
+                               label="Plasma")
+                ax1[0, 1].set_ylabel('[$cm^{-3}$]')
+            else:   
+                ax1[0, 1].plot(time_average, reservoirs["particles_in_plasma"],
+                           label="Plasma")
+                ax1[0, 1].set_ylabel('[#]')
+
+            if self.namelist["phys_volumes"]:
+                ax1[0, 2].plot(time_average, reservoirs["particle_density_in_divertor"],
+                               label="Div. reservoir")
+                if self.namelist["pump_chamber"]:
+                    ax1[0, 2].plot(time_average, reservoirs["particle_density_in_pump"],
+                               label="Pump reservoir")
+                ax1[0, 2].set_ylabel('[$cm^{-3}$]')     
+            else:
+                ax1[0, 2].plot(time_average, reservoirs["particles_in_divertor"],
+                           label="Div. reservoir")
+                if self.namelist["pump_chamber"]:
+                    ax1[0, 2].plot(time_average, reservoirs["particles_in_pump"],
+                           label="Pump reservoir")
+                ax1[0, 2].set_ylabel('[#]')
+
+            ax1[1, 0].plot(time_average, reservoirs["edge_loss"], label="Edge loss to wall")
+            ax1[1, 0].plot(time_average, reservoirs["limiter_loss"], label="Limiter loss to wall")
+            ax1[1, 0].set_ylabel('[$s^{-1}$]')  
+
+            ax1[1, 1].plot(time_average, reservoirs["wall_recycling"], label="Wall rec. rate")
+            ax1[1, 1].set_ylabel('[$s^{-1}$]')  
+
+            ax1[1, 2].plot(time_average, reservoirs["particles_stuck_at_wall"], label="Part. stuck at wall")
+            ax1[1, 2].plot(time_average, reservoirs["particles_retained_at_wall"],
+                label="Part. retained at wall")
+            ax1[1, 2].set_ylabel('[#]')
+
+            ax1[2, 0].plot(time_average, reservoirs["parallel_loss"], label="Par. loss to divertor")
+            ax1[2, 0].plot(time_average, reservoirs["divertor_backflow"],
+                           label="Divertor backflow rate")
+            if self.namelist["pump_chamber"]:
+                ax1[2, 0].plot(time_average, reservoirs["pump_leakage"],
+                           label="Pump leakage rate")
+            ax1[2, 0].set_ylabel('[$s^{-1}$]')
+
+            for aa in ax1.flatten()[:7]:
+                aa.legend(loc="best").set_draggable(True)
+
+            for ii in [0, 1, 2]:
+                ax1[2, ii].set_xlabel("Time [s]")
+            ax1[2, 0].set_xlim(time_average[[0, -1]])
+
+            # ----------------------------------------------------------------
+            # plot the impurity radiation, if present:
+            if "impurity_radiation" in reservoirs:
```

Comment:
@antonellozito why did you decide to create a third axis? I quite like the idea of having impurity radiation on the same axes when looking at a full run...

Reply:
@fsciortino this is actually related to some changes that I have in mind but that still do not exist. Basically, what I want to do later is to distinguish between a main wall reservoir, and a divertor wall reservoir, distinction that now is not present. In this sense, I thought "aesthetically" to organize the plots, in the reservoir windows, as:

- Second row: everything related to the main wall, i.e.: [1,0]: edge/limiter losses, [1,1]: recycling rate from main wall, [1,2]: dynamic main wall reservoir content.
- Third row: everything related to the divertor wall, i.e.: [2,0]: parallel losses + backflow towards core, [2,1]: recycling rate from divertor wall, [2,2]: dynamic divertor wall reservoir content.

Since the divertor wall reservoir doesn't exist yet, there is nothing now to fill the final two axis in the third row, but I kept that empty in order to "fill" them later. With this idea, as the only remaining plot would have been the impurity radiation one (which does not even appear by default), I though about putting it in a separated axis.
This is the rationale behind my choice, but I am open to discuss it!